### PR TITLE
Add MAAS Metric for rabbit queue length

### LINKED ISSUE #633: Rabbitmq Queue Length Metric required
"For the RabbitMQ queue length, the scenario we want to detect is a queue growing due to a failed consumer or consumers that can’t keep up with load."

"Support confirmed (via @bjoern) that monitoring and alerting on the following single metric would be sufficient: ((Total message count across all queues) – (messages in notifications queue))"

Further details: 
https://trello.com/c/widYAV5F/35-add-new-monitor-rabbitmq-queue-length

### PR DESCRIPTION
**Pull request by hughsaunders**
Add MAAS Metric for rabbit queue length

* Adds a single metric: queued_messages_excluding_notifications
* Adds an alarm with configurable threshold (default 100)
* Refactors the rabbitmq plugin to reduce duplicated code

Closes #633
Trello: https://trello.com/c/widYAV5F/35-add-new-monitor-rabbitmq-queue-length

### CODE REVIEW

**Comment by hughsaunders** [comment]
Removed erroneous SHA bump from commit

**Comment by git-harry** [comment]
@hughsaunders rebase required

File: maas/plugins/rabbitmq_status.py [hunk 1]
```diff
@@ -85,79 +86,91 @@ def parse_args():
...
             status_err('cluster too small')
         if not is_cluster_member:
             status_err('{0} not a member of the cluster'.format(name))
 
+
+def _get_queue_metrics(session, metrics, host, port):
+    response = _get_rabbit_json(session, QUEUES_URL % (host, port))
+    all_queued_messages = sum([q['messages'] for q in response])
+    notification_messages = sum([q['messages'] for q in response
+                                if q['name'].startswith('notification.')])
```
Comment (mancdaz) [inline maas/plugins/rabbitmq_status.py]: this will fail to pick up the correct queue as the notifications queue is ```notifications.info```

**Comment by mancdaz** [comment]
:-1: this doesn't currently work

**Comment by git-harry** [comment]
-1 I'm opposed to the metric queued_messages_excluding_notifications. The commit provides no details but my understanding is that this is being done because the notifications queue is not consumed. Instead of working round the issue we should solve it [1]. This prevents rabbitmq from consuming unnecessary resources when the queue is not required.

[1] https://github.com/openstack/oslo.messaging/blob/3a5db723aa0613a99df6d95eecf6d8cecf2ca79d/doc/source/FAQ.rst

**Comment by mancdaz** [comment]
@hughsaunders did you mean to include a sha bump for rpc-openstack in this commit?

**Comment by hughsaunders** [comment]
Removed sha bump. This is now tested and ready for review.

**Comment by hughsaunders** [comment]
updated metric names in readme

**Comment by mancdaz** [comment]
@git-harry I agree we should turn off notifications if we don't want them.  Turns out we already do, except for neutron. So I've started the review process to get that fixed:

https://review.openstack.org/#/c/252416/

In the interim, and for current deployments, I think we need to go with the current approach of generating a metric that has the notifications messages subtracted.

I've tested on a rpc-openstack master deployment and this current patchset works for me

Therefore, for me right now,  :+1:

**Comment by andymcc** [comment]
Works for me - tested both alarms + new metrics.

**Pull request merged**